File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int sum = 0;

        Arrays.sort(people);

        Deque <Integer> deque = new ArrayDeque<>();
        for (int p : people) {
            deque.add(p);
        }
        
        while (!deque.isEmpty()){
            sum = deque.pollLast();
            if (!deque.isEmpty() && sum + deque.peekFirst() <= limit){
                deque.pollFirst();
            }
            answer+=1;
        }
 
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque
def solution(people, limit):
    people.sort()
    pList = deque(people)
    answer = 0
    while len(pList) != 0:
        sum = pList.pop()
        if len(pList) != 0 and sum + pList[0] <= limit:
            pList.popleft()
        answer += 1
    return answer